test/remote/gateways/remote_paymill_test.rb

```diff
@@ -2,71 +2,83 @@
 
 class RemotePaymillTest < Test::Unit::TestCase
   def setup
-    @gateway = PaymillGateway.new(fixtures(:paymill))
+    params = fixtures(:paymill)
+    @gateway = PaymillGateway.new(public_key: params.delete(:public_key), private_key: params.delete(:private_key))
 
     @amount = 100
     @credit_card = credit_card('5500000000000004')
     @declined_card = credit_card('5105105105105100', month: 5, year: 2020)
+
+    params['presentation.amount3D'.intern] = @amount
+    params['channel.id'.intern] = @gateway.options[:public_key]
+
+    uri = URI.parse("#{params.delete('tokenization.url'.intern)}?#{params.to_a.map{ |param| param.join('=') }.join('&')}")
+    https = Net::HTTP.new(uri.host, uri.port)
+    https.use_ssl = true
+    request = Net::HTTP::Get.new(uri.request_uri)
+    request.basic_auth(@gateway.options[:private_key], "")
+    response = https.request(request)
+    @token = response.body.match('tok_[a-z|0-9]+')[0]
   end
 
   def test_successful_purchase
     assert response = @gateway.purchase(@amount, @credit_card)
     assert_success response
-    assert_equal 'General success response.', response.message
+    assert_equal 'Operation successful', response.message
   end
 
   def test_successful_purchase_with_token
-    assert response = @gateway.purchase(@amount, '098f6bcd4621d373cade4e832627b4f6', { description: 'from active merchant' } )
+    assert response = @gateway.purchase(@amount, @token, { description: 'from active merchant' } )
```

Comment:
These changes look good except for this - the description you're passing in isn't going anywhere anymore. Should it still be getting included, or should this be removed?